### PR TITLE
call-back feature

### PR DESCRIPTION
**Pull request by floridemai**
I did leave some notes in the code.

Some additional, more general, notes:
- i've tried to not remove things and just add extra (e.g. phone-number & phone-number-us component)
- FollowUp was the previous implementation, CallBack is the current one - in terms of naming
- renamed storage keys prefix
- Markdown component needs an update across the board (margins mainly)
- Close contact text is not dynamic (from api), but loaded from the assets
- Should we have a "card button"?

**PLEASE** review, test and comment so that we can have this as close to the final version as possible. Thank you.

### CODE REVIEW

File: src/assets/country-codes.ts [hunk 1]
```diff
@@ -4,17 +4,22 @@ export interface CountryCode {
   iso: string;
 }
 
+// Codes of countries to push to start of list, in order
```
Comment (floridemai) [inline src/assets/country-codes.ts:7]: Code is not needed for the current implementation, but it's an improvement

File: src/components/atoms/toast.tsx [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
-  TextStyle,
-} from 'react-native';
+import React, {ReactNode} from 'react';
+import {StyleSheet, View, ViewStyle, Text, TextStyle} from 'react-native';
 
 import {Markdown} from './markdown';
 
 import {text, colors} from 'theme';
 
+export type ToastType = 'default' | 'success' | 'warning';
```
Comment (floridemai) [inline src/components/atoms/toast.tsx:8]: The toast component feels like it needs a refactor to support just success and warning implementations (we don't have any other type of toasts in the app)

File: src/components/molecules/country-code-dropdown.tsx [hunk 3]
```diff
@@ -2,15 +2,21 @@ import React, {FC, useState} from 'react';
 import {StyleSheet, View, Text} from 'react-native';
 import {useTranslation} from 'react-i18next';
 
-import countryCodes from 'assets/country-codes';
 import {Dropdown} from 'components/atoms/dropdown';
+
 import {text, colors} from 'theme';
 
+import countryCodes, {primaryCodes, CountryCode} from 'assets/country-codes';
```
Comment (floridemai) [inline src/components/molecules/country-code-dropdown.tsx:9]: Code irrelevant to the current implementation

File: src/components/molecules/result-card.tsx [hunk 4]
```diff
@@ -1,132 +1,72 @@
-import React from 'react';
-import {StyleSheet, Text, View} from 'react-native';
-import {useNavigation} from '@react-navigation/native';
-import {useTranslation} from 'react-i18next';
-import * as WebBrowser from 'expo-web-browser';
+import React, {FC} from 'react';
```
Comment (floridemai) [inline src/components/molecules/result-card.tsx:1]: This result card is a **brand new card** that's supposed to be used in all places where the screen is just a confirmation. The current implementation is a fragile one, which only covers the success page of the callback feature.

The previous "result-card" was renamed to "symptom-checker-result-card" and it is untocuhed.

File: src/components/organisms/phone-number-us.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
+import React, {FC, useState, useRef} from 'react';
```
Comment (floridemai) [inline src/components/organisms/phone-number-us.tsx:1]: A copy of the previous phone-number component that has the country code removed. Also a new asset key was created for this.

File: src/services/api/index.ts [hunk 6]
```diff
@@ -388,3 +392,22 @@ export async function saveMetric({event = ''}) {
     return false;
   }
 }
+
+export const uploadCallBackData = async (
+  callBackData: CallBackData
+): Promise<UploadResponse> => {
+  const resp = (await request(`${urls.api}/callback`, {
```
Comment (floridemai) [inline src/services/api/index.ts:399]: If this needs to request to a different address, let me know.

File: src/App.tsx [hunk 7]
```diff
@@ -50,7 +49,12 @@ import {SymptomChecker} from 'components/views/symptom-checker';
 import {SymptomsHistory} from 'components/views/symptoms-history';
 import {ContactTracing} from 'components/views/contact-tracing';
 import {CountyBreakdown} from 'components/views/county-breakdown';
+import {CloseContactInfo} from './components/views/close-contact-info';
 import {CloseContact} from 'components/views/close-contact';
+import {CloseContactRequiredAge} from './components/views/close-contact-required-age';
+import {CloseContactUnderAge} from './components/views/close-contact-under-age';
```
Comment (ckiss) [inline src/App.tsx]: you should use alias import

File: src/components/views/close-contact-info.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React, {FC} from 'react';
+import {useTranslation} from 'react-i18next';
+
+import {Markdown} from 'components/atoms/markdown';
+import {Scrollable} from 'components/templates/scrollable';
+
+export const CloseContactInfo: FC<any> = () => {
```
Comment (ckiss) [inline src/components/views/close-contact-info.tsx]: nit: I think is better to remove <any>

File: src/components/views/close-contact-under-age.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  strong: {
+    ...text.largeBold
+  },
+  block: {
+    marginBottom: 32
+  }
+};
+
+export const CloseContactUnderAge: FC<any> = () => {
```
Comment (ckiss) [inline src/components/views/close-contact-under-age.tsx]: same about any

**Review by ckiss — APPROVED** [review]
LGTM, small comments

File: src/components/views/call-back.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  const errorToast = error && (
+    <Toast
+      type="warning"
+      icon={<AppIcons.Alert width={24} height={24} />}
+      message={t(`callBack:error:${error}`)}
+    />
+  );
+
+  if (!callBackIsQueued) {
+    <KeyboardScrollable toast={errorToast} scrollViewRef={scrollViewRef}>
```
Comment (ckiss) [inline src/components/views/call-back.tsx]: missing a `return`

File: src/components/views/close-contact.tsx [hunk 11]
```diff
@@ -1,71 +1,69 @@
...
+      <Markdown markdownStyles={markdownStyles}>
+        {t('closeContact:intro')}
+      </Markdown>
+      {callBackIsQueued ? (
         <>
-          <Card icon={<BubbleIcons.PhoneCall width={56} height={56} />}>
-            <Text style={text.largeBlack}>{t('closeContact:callBack')}</Text>
-            <Text style={styles.notice}>{t('closeContact:callBackQueue')}</Text>
+          <Text style={text.largeBold}>
+            {t('closeContact:callQueued:title')}
```
Comment (ckiss) [inline src/components/views/close-contact.tsx:45]: this is not defined in the resources

File: src/components/views/close-contact-info.tsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+  const {t} = useTranslation();
+
+  return (
+    <Scrollable heading={t('closeContactInfo:title')}>
+      <Markdown>{t('closeContactInfo:info')}</Markdown>
```
Comment (ckiss) [inline src/components/views/close-contact-info.tsx:12]: this doesn't look as per designed (missing numbered list)

File: src/components/views/contact-tracing/index.tsx [hunk 13]
```diff
@@ -113,7 +115,11 @@ export const ContactTracing = ({navigation}) => {
         <>
           <Spacing s={16} />
           <Card
-            onPress={() => navigation.navigate('closeContact', {info: true})}
+            onPress={() =>
+              navigation.navigate(
+                !hasCloseContact ? 'closeContact' : 'closeContactInfo'
```
Comment (ckiss) [inline src/components/views/contact-tracing/index.tsx]: this should be reversed

**Review by ShaunBaker — APPROVED** [review]

File: src/components/views/call-back.tsx (hunk 10 above)
Comment (floridemai) [reply src/components/views/call-back.tsx]: Nice catch. Thanks!

File: src/components/views/close-contact.tsx (hunk 11 above)
Comment (floridemai) [reply src/components/views/close-contact.tsx:45]: There is not a design for it. I'll merge it like this to force it being updated

File: src/components/views/contact-tracing/index.tsx (hunk 13 above)
Comment (floridemai) [reply src/components/views/contact-tracing/index.tsx]: 😬...

**Pull request merged**